OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -1033,11 +1033,18 @@ void testInputOutputConnections()
 
     // do any other input/output connections
     foo1->updInput("input1").connect(bar->getOutput("PotentialEnergy"));
+
     // Must throw an exception since subState0 is not a state of internalSub.
-    ASSERT_THROW(OpenSim::Exception, 
-        foo2->updInput("input1").connect(world.getOutput("./internalSub/subState0")));
+    ASSERT_THROW(OpenSim::Exception,
+        foo2->updInput("input1").connect(world.getComponent("./internalSub").getOutput("subState0")));
```

Comment:
I haven't been able to find where the leading period is handled in `getComponent()`. The comment indicates that this line should throw an exception because subState0 is not a state of internalSub, but is the exception being thrown because of the "."? Perhaps more specific exceptions should be defined so we know why they're being thrown. See [the top of Component.h](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Component.h#L63), for example.